.github/workflows/clang.yaml

```diff
@@ -0,0 +1,88 @@
+name: Build with Clang/LLVM
+
+on: pull_request_target
+
+jobs:
+  clang-build-cancel:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+  clang-build:
+    runs-on: ubuntu-latest
+    needs: clang-build-cancel
+    container:
+      image: zephyrprojectrtos/ci:v0.17.0-rc1
+      options: '--entrypoint /bin/bash'
+    strategy:
+      fail-fast: false
+      matrix:
+        subset: [1, 2, 3, 4, 5, 6, 7 , 8, 9, 10]
+    env:
+      ZEPHYR_SDK_INSTALL_DIR: /opt/toolchains/zephyr-sdk-0.12.4
+      CLANG_ROOT_DIR: /usr/lib/llvm-12
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+      - name: Update PATH for west
+        run: |
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: west setup
```

Comment:
We should create an action for this, it's being duplicated all over the place. I will take a look